src/components/playback/playback-control.tsx

```diff
@@ -0,0 +1,204 @@
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import Slider from "rc-slider";
+import classNames from "classnames";
+import { Instance } from "mobx-state-tree";
+import { observer } from "mobx-react";
+import { useUIStore } from "../../hooks/use-stores";
+import { TreeManager } from "../../models/history/tree-manager";
+import { PlaybackMarkerToolbar } from "./marker-toolbar";
+import Marker from "../../clue/assets/icons/playback/marker.svg";
+import PlayButton from "../../clue/assets/icons/playback/play-button.svg";
+import PauseButton from "../../clue/assets/icons/playback/pause-button.svg";
+
+import "./playback-control.scss";
+
+export interface IMarkerProps {
+  id: number;
+  location: number;
+}
+
+interface IProps {
+  treeManager: Instance<typeof TreeManager>;
+}
+
+export const PlaybackControlComponent: React.FC<IProps> = observer((props: IProps) => {
+  const { treeManager } = props;
+  const { activeNavTab } = useUIStore();
+  const [sliderPlaying, setSliderPlaying] = useState(false);
+  const sliderContainerRef = useRef<HTMLDivElement>(null);
+  const railRef = useRef<HTMLDivElement>(null);
+  const [markerSelected, setMarkerSelected] = useState(false);
+  const [addMarkerButtonSelected, setAddMarkerButtonSelected] = useState(false);
+  const [markers, setMarkers] = useState<IMarkerProps[]>([]);
+  const [selectedMarkers, ] = useState<IMarkerProps[]>([]);
+  const history = treeManager.document.history;
+  const [sliderValue, setSliderValue] = useState(history.length);
+  const eventAtCurrentIndex = treeManager.currentHistoryIndex === 0
+                                ? undefined
+                                : treeManager.getHistoryEntry(treeManager.currentHistoryIndex - 1);
+  const eventCreatedTime = eventAtCurrentIndex?.created;
+  const playbackDisabled = treeManager.currentHistoryIndex === undefined || sliderValue === history.length;
+
+  const handlePlayPauseToggle = useCallback((playing?: boolean) => {
+                                  if (playing) {
+                                    setSliderPlaying(playing);
+                                  } else {
+                                    setSliderPlaying(!sliderPlaying);
+                                  }
+                                },[sliderPlaying]);
+
+  // If our slider value is ever beyond the end of the known history, rein it in.
+  // This happens on initial load as seemingly spurious entries are automatically
+  // removed from the history.
+  useEffect(() => {
+    if (sliderValue > history.length) {
+      setSliderValue(history.length);
+    }
+  },[history.length, sliderValue]);
+
+  useEffect(() => {
+    if (sliderPlaying) {
+      const slider = setInterval(()=>{
```

Comment:
It would be more clear if you used `setTimeout` here instead.  When the the delayed function calls setSliderValue, it will trigger a re-render. And on that re-render sliderValue will be changed so the useEffect be triggered again. This means the original dispose function of the old effect will be called. Which means clearInterval will be called first and then setInterval will be called again. So in most cases the setInterval will only ever run one time.

The case where setInterval will run more than once is if goToHistoryEntry takes a really long time to do its thing.  In that case setSliderValue will not be called before the second interval.  But this is bad because it means goToHistoryEntry will be called a second time while it is still trying to handle the first time. Using setTimeout should prevent this from happening.